### PR TITLE
Adding godoc support for ref docs

### PR DESCRIPTION
**Pull request by modnarb**
Formatting edits, etc.

### CODE REVIEW

File: dialog/doc.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+
+// Package dialog provides functionality for client-service speech and activity interactions
```
Comment (glecaros) [inline dialog/doc.go:4]: Shouldn't we talk about the whole virtual assistant stuff here?

File: audio/doc.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+
+// Package audio implements the configuration, input/output streams, and related utilities for audio
```
Comment (glecaros) [inline audio/doc.go:4]: Maybe say something different from implements... maybe provide?

Comment (glecaros) [reply audio/doc.go:4]: Maybe say audio configuration

File: go.mod [hunk 3]
```diff
@@ -1,3 +1,5 @@
 module github.com/Microsoft/cognitive-services-speech-sdk-go
 
 go 1.13
+
+require golang.org/x/tools v0.0.0-20200414211825-33e937220d8f // indirect
```
Comment (glecaros) [inline go.mod:5]: maybe we shoudn't ship this change

File: go.sum [hunk 4]
```diff
@@ -0,0 +1,20 @@
+github.com/yuin/goldmark v1.1.27/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
```
Comment (glecaros) [inline go.sum:1]: or this

File: samples/dialog_service_connector/from_push_audio_stream.go [hunk 5]
```diff
@@ -1,7 +1,7 @@
-//
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
-//
+
+// Package dialog_service_connector sample for single-turn recognition from a PushAudioInputStream
```
Comment (glecaros) [inline samples/dialog_service_connector/from_push_audio_stream.go:4]: The description does not seem accurate

File: samples/dialog_service_connector/start_keyword_listening.go [hunk 6]
```diff
@@ -1,7 +1,7 @@
-//
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
-//
+
+// Package dialog_service_connector sample for keyword recognition
```
Comment (glecaros) [inline samples/dialog_service_connector/start_keyword_listening.go:4]: This should probably not be here?

File: samples/go.mod [hunk 7]
```diff
@@ -1,7 +1,8 @@
 module github.com/Microsoft/cognitive-services-speech-sdk-go/samples
 
 require (
-    github.com/Microsoft/cognitive-services-speech-sdk-go v1.11.0-alpha1
+	github.com/Microsoft/cognitive-services-speech-sdk-go v1.11.0-alpha1
+	golang.org/x/tools v0.0.0-20200414211825-33e937220d8f // indirect
```
Comment (glecaros) [inline samples/go.mod:5]: I would prefer if we don't add this dependency here .

File: speech/doc.go [hunk 8]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+
+// Package speech implements speech recognizers, along with their related configuration and event objects
```
Comment (glecaros) [inline speech/doc.go:4]: Implements probably not the best word here.

File: samples/main.go [hunk 9]
```diff
@@ -1,7 +1,7 @@
-//
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
-//
+
+// Package samples demonstrates usages for the speech recognizer and dialog service connector
```
Comment (glecaros) [inline samples/main.go:4]: Wrong package

File: samples/go.sum [hunk 10]
```diff
@@ -1,2 +1,22 @@
 github.com/Microsoft/cognitive-services-speech-sdk-go v1.11.0-alpha1 h1:dOPa6u1Y8q25uWBayW47GFMAG2SWNx9vA2vS0k7CHEg=
 github.com/Microsoft/cognitive-services-speech-sdk-go v1.11.0-alpha1/go.mod h1:ct4bG95K1Lu/c5y60PVGI1XOjo9aAcl80DD5dvu6zsg=
+github.com/yuin/goldmark v1.1.27/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
```
Comment (glecaros) [inline samples/go.sum:3]: same as go.mod

File: dialog/doc.go (hunk 1 above)
Comment (modnarb) [reply dialog/doc.go:4]: How about this?

// Package dialog provides functionality for creating custom voice assistant applications and managing the related interaction flow

File: go.sum (hunk 4 above)
Comment (modnarb) [reply go.sum:1]: wasn't sure about these, but I'll remove

File: samples/dialog_service_connector/start_keyword_listening.go (hunk 6 above)
Comment (modnarb) [reply samples/dialog_service_connector/start_keyword_listening.go:4]: I'll create a standalone doc.go with this package name and combine the descriptors

File: samples/go.sum (hunk 10 above)
Comment (modnarb) [reply samples/go.sum:3]: so, remove this line or all below as well?

**Pull request merged**